tools/build-tools/src/test/typeValidator/classValidator.spec.ts

```diff
@@ -49,92 +49,503 @@ describe("Class", () => {
+            }
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // setters are treated the same as public methods
+        // adding/removing one is a breaking change
+        it("added a setter", () => {
+            const classOld =
+            `
+            export class TestClass {
+                private _testProperty = "sussy baka";
+            }
+            `;
+
+            const classNew =
+            `
+            export class TestClass {
+                private _testProperty = "sussy baka";
+                set setTestProperty(newTestProperty: string) {
+                    this._testProperty = newTestProperty;
+                }
+            }
+            `;
 
-        let increment = checkIncrement(classOld, classNew);
-        assert(increment == BreakingIncrement.major);
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // adding a new required method parameter is a breaking change
+        it("added a new required method param", () => {
+            const classOld =
+            `
+            export class TestClass {
+                public testMethod() { return false; }
+            }
+            `;
+
+            const classNew =
+            `
+            export class TestClass {
+                public testMethod(newProp: boolean) { return false; }
+            }
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // scenario: added new default param (method signature changed but won't break existing code)
+        // expected result: minor breaking change
+        it("new default value added", () => {
+            const classOld =
+            `
+            export class TestClass {
+                public testMethod() { return "afg"; }
+            }
+            `;
+
+            const classNew =
+            `
+            export class TestClass {
+                public testMethod(param1 : string = "afg") { return "afg"; }
+            }
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.minor);
+        });
+
+        // scenario: new version changes the param to a different type.
+        // expected result: A major breaking change
+        it("default value type changed", () => {
+            const classOld =
+            `
+            export class TestClass {
+                public testMethod(param1 : string = "afg") { return "afg"; }
+            }
+            `;
+
+            const classNew =
+            `
+            export class TestClass {
+                public testMethod(param1 : boolean = false) { return "afg"; }
```

Comment:
>afg [](http://example.com/codeflow?start=70&length=3)

nit: I like to use strings like "value-does-not-matter" or something.

Reply:
Or a constant?

Reply:
uh oh not sure you'll like my other naming choices later in the tests then...

Reply:
hahaha, it's not a big deal.